tests/8.LogicSig.ts

```diff
@@ -0,0 +1,1487 @@
+              'base64'
+            )
+          ),
+        };
+        testSign(lsig, sender, expected);
+      });
+
+      it('should match expected when sender is not LogicSig address', () => {
+        const sender = otherAddr;
+        const expected = {
+          txID: 'DRBC5KBOYEUCL6L6H45GQSRKCCUTPNELUHUSQO4ZWCEODJEXQBBQ',
+          blob: new Uint8Array(
+            Buffer.from(
+              'g6Rsc2lngqNhcmeSxAEBxAICA6FsxAUBIAEBIqRzZ25yxCD2di2sdbGZfWwslhgGgFB0kNeVES/+f7dgsnOK+cfxraN0eG6Ko2FtdM0TiKNmZWXOAANPqKJmds4ADtbco2dlbq10ZXN0bmV0LXYzMS4womdoxCAmCyAJoJOohot5WHIvpeVG7eftF+TYXEx4r7BFJpDt0qJsds4ADtrEpG5vdGXECLRReTn8+tJxo3JjdsQgtMYiaKTDNVD1im3UuMojnJ8dELNBqn4aNuPOYfv8+Yqjc25kxCC0xiJopMM1UPWKbdS4yiOcnx0Qs0Gqfho2485h+/z5iqR0eXBlo3BheQ==',
+              'base64'
+            )
+          ),
+        };
+        testSign(lsig, sender, expected);
+      });
+    });
+
+    describe('single sig', () => {
+      const account = algosdk.mnemonicToSecretKey(
+        'olympic cricket tower model share zone grid twist sponsor avoid eight apology patient party success claim famous rapid donor pledge bomb mystery security ability often'
+      );
+      const lsig = algosdk.makeLogicSig(program, args);
+      lsig.sign(account.sk);
+
+      it('should match expected when sender is LogicSig address', () => {
+        const sender = account.addr;
+        const expected = {
+          txID: 'EZB2N2TEFR5OOL76Z46ZMRUL3ZQQOYKRFIX6WSHQ5FWESHU4LZPA',
+          blob: new Uint8Array(
+            Buffer.from(
+              'gqRsc2lng6NhcmeSxAEBxAICA6FsxAUBIAEBIqNzaWfEQD4FPTlN+xK8ZXmf6jGKe46iUYtVLIq+bNenZS3YsBh+IQUtuSRiiRblYXTNDxmsuWxFpCmRmREd5Hzk/BLszgKjdHhuiqNhbXTNE4ijZmVlzgADT6iiZnbOAA7W3KNnZW6tdGVzdG5ldC12MzEuMKJnaMQgJgsgCaCTqIaLeVhyL6XlRu3n7Rfk2FxMeK+wRSaQ7dKibHbOAA7axKRub3RlxAi0UXk5/PrScaNyY3bEILTGImikwzVQ9Ypt1LjKI5yfHRCzQap+GjbjzmH7/PmKo3NuZMQgXmdPHAru7DdxiY9hx2/10koZeT4skfoIUWJj44Vz6kKkdHlwZaNwYXk=',
+              'base64'
+            )
+          ),
+        };
+        testSign(lsig, sender, expected);
+      });
+
+      it('should throw an error when sender is not LogicSig address', () => {
+        const txn = algosdk.makePaymentTxnWithSuggestedParamsFromObject({
+          from: otherAddr,
+          to: otherAddr,
+          amount: 5000,
+          suggestedParams: {
+            flatFee: true,
+            fee: 217000,
+            firstRound: 972508,
+            lastRound: 973508,
+            genesisID: 'testnet-v31.0',
+            genesisHash: 'JgsgCaCTqIaLeVhyL6XlRu3n7Rfk2FxMeK+wRSaQ7dI=',
+          },
+          note: new Uint8Array([180, 81, 121, 57, 252, 250, 210, 113]),
+        });
+
+        assert.throws(
+          () => algosdk.signLogicSigTransaction(txn, lsig),
+          (err) =>
+            err.message ===
+            'Logic signature verification failed. Ensure the program and signature are valid.'
```

Comment:
If only checking for equality, the following `assert.throws` overload is more concise:

```
assert.throws(() => algosdk.signLogicSigTransaction(txn, lsig), 'Logic signature verification failed. Ensure the program and signature are valid.')
```

Reply:
Good point, I originally thought that syntax wasn't valid but I think I had other mistakes that prevented it from working

Reply:
Actually I still get this error when changing this to the above: `TypeError [ERR_AMBIGUOUS_ARGUMENT]: The "error/message" argument is ambiguous. The error message "Logic signature verification failed. Ensure the program and signature are valid." is identical to the message.`

Reply:
According to https://nodejs.org/api/assert.html#assert_assert_throws_fn_error_message a string isn't allowed as the second argument. I could change it to a RegExp to get it to work, but it's not clear to me whether that would be better than the current code.